sql/core/src/main/scala/org/apache/spark/sql/execution/limit.scala

```diff
@@ -240,7 +298,7 @@ case class TakeOrderedAndProjectExec(
           readMetrics)
       }
       singlePartitionRDD.mapPartitions { iter =>
-        val topK = Utils.takeOrdered(iter.map(_.copy()), limit)(ord)
+        val topK = Utils.takeOrdered(iter.map(_.copy()), limit + offset)(ord).drop(offset)
```

Comment:
shall we also add an if-else here and only call `.drop` if `offset > 0`?

Reply:
OK